### DISCUSSION TITLE
Is it possible to obtain branch information from the Github Page configuration, thereby omitting the branch parameter

### DISCUSSION THREAD

**Discussion by kevinLiJ** [JamesIves/github-pages-deploy-action#552] [Ideas]
I have configured the branch information in the configuration of GitHub Pages, why do I need to pass in the branch parameter in the action?

**Comment by Pike** [comment]
The API seems to exist, https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#get-a-github-pages-site.

FWIW, it's not exposed on the graphql API, I searched for that for a while.

There seem to be  (undocumented) permission requirements for that API, so it's a bit hard to tell if it could be used as part of an action.

It would be a good default for both `branch` and `target_folder`, I guess?

**Reply by Pike** [reply]
I did look at this for a moment, and it's easier said than done. Not that it's really hard, but all the `getInput` calls in `constants.ts` are independent of each other right now.

For the API, we'd first need the token, then get the rest api, and then call the rest api. And we'd do that for two options, but only for the default. And all of that would be `async`. So that'd be a quite a few changes rattling through probably a number of places.
